spot-ml/src/main/scala/org/apache/spot/dns/model/DNSSuspiciousConnectsModel.scala

```diff
@@ -164,22 +167,70 @@ object DNSSuspiciousConnectsModel {
 
     // create quantile cut-offs
 
-    val timeCuts = Quantiles.computeDeciles(totalDataDF.select(UnixTimestamp).rdd.
-      map({ case Row(unixTimeStamp: Long) => unixTimeStamp.toDouble }))
-
-    val frameLengthCuts = Quantiles.computeDeciles(totalDataDF.select(FrameLength).rdd
-      .map({ case Row(frameLen: Int) => frameLen.toDouble }))
-
-    val domainStatsDF = createDomainStatsDF(sparkContext, sqlContext, countryCodesBC, topDomainsBC, userDomain, totalDataDF)
-
-    val subdomainLengthCuts = Quantiles.computeQuintiles(domainStatsDF.filter(SubdomainLength + " > 0")
-      .select(SubdomainLength).rdd.map({ case Row(subdomainLength: Int) => subdomainLength.toDouble }))
-
-    val entropyCuts = Quantiles.computeQuintiles(domainStatsDF.filter(SubdomainEntropy + " > 0")
-      .select(SubdomainEntropy).rdd.map({ case Row(subdomainEntropy: Double) => subdomainEntropy }))
-
-    val numberPeriodsCuts = Quantiles.computeQuintiles(domainStatsDF.filter(NumPeriods + " > 0")
-      .select(NumPeriods).rdd.map({ case Row(numberPeriods: Int) => numberPeriods.toDouble }))
+    val timeCuts =
+      Quantiles.computeDeciles(totalRecords
+        .select(UnixTimestamp)
+        .rdd
+        .flatMap({ case Row(unixTimeStamp: Long) => {
```

Comment:
maybe not for this checkin, but it looks like there is a repeated pattern in all this could that could somehow be reused